Revise unifying techniques intro (#46)
greenelab/knowledge-graph-review: content/04.network_applications.md

## Applying Knowledge Graphs to Biomedical Challenges

Knowledge graphs can help researchers tackle many biomedical problems such as finding new treatments for existing drugs [@doi:10.7554/eLife.26726], aiding efforts to diagnose patients [@arxiv:1611.07012] and identfying associations between diseases and biomolecules [@doi:10.1155/2017/2498957].
In many cases, solutions rely on representing knowledge graphs in a low dimensional space, which is a process called representation learning.

> This space preserves a knowledge graph's local and/or global structure and is straightforward to utilize in efforts that apply machine learning methods to make predictions.

Reviewer: ```suggestion
This goal of this process is to retain and encode the local and/or global structure of a knowledge graph that are relevant to the problem while transforming the graph into a representation that can be readily used with machine learning methods to build predictors
```
Maybe this will help with the reviewer?

> In the following sections we review methods that construct this low dimensional space (Unifying Relational Learning Techniques) and discuss applications that use this space to solve biomedical problems (Unifying Applications).

Reviewer: Where does the name "Relational Learning Techniques" come from? Is it widely used in the field?

Author: Relational learning came from reviewer 2's comment and from [the paper](https://arxiv.org/pdf/1503.00759.pdf) reviewer 2 mentioned. This term is used in the field to represent methods that work on knowledge graphs (relational data). Upon glancing at the paper, it seems that this term comes from a more statistical point of view. 

After some internal debate, I think the term "representational learning" would be more appropriate as techniques mentioned in this paper focus on low dimensional knowledge graph representations. This term is also widely used in the field, but is more associated with low dimensional embeddings.

## Applying Knowledge Graphs to Biomedical Challenges

Knowledge graphs can help researchers tackle many biomedical problems such as finding new treatments for existing drugs [@doi:10.7554/eLife.26726], aiding efforts to diagnose patients [@arxiv:1611.07012] and identfying associations between diseases and biomolecules [@doi:10.1155/2017/2498957].
In many cases, solutions rely on representing knowledge graphs in a low dimensional space, which is a process called representational learning.

> This goal of this process is to retain and encode the local and/or global structure of a knowledge graph that are relevant to the problem while transforming the graph into a representation that can be readily used with machine learning methods to build predictors

Reviewer: ```suggestion
This goal of this process is to retain and encode the local and/or global structure of a knowledge graph that are relevant to the problem while transforming the graph into a representation that can be readily used with machine learning methods to build predictors.
```

In the following sections we review methods that construct this low dimensional space (Unifying Representational Learning Techniques) and discuss applications that use this space to solve biomedical problems (Unifying Applications).

### Unifying Representational Learning Techniques

Mapping high dimensional data into a low dimensional space greatly improves modeling performance in fields such as natural language processing [@arxiv:1301.3781; @arxiv:1310.4546] and image analysis [@arxiv:1512.03385].
The success of these approaches served as rationale for a sharper focus on representing knowledge graphs in a low dimensional space [@arxiv:1709.05584].

> This space captures the essence of a knowledge graph in the form of dense vectors, which are lists of continuous values of user-specified length; however, some algorithms pre-define the length of each vector [@doi:10.1007/BF02288367; @doi:10.1162/089976603321780317].

Reviewer: ```suggestion
Methods of this class are designed to capture the essence of a knowledge graph in the form of dense vectors [@doi:10.1007/BF02288367; @doi:10.1162/089976603321780317].
```

> These vectors are often assigned to nodes in a graph, but edges can be assigned as well.

Reviewer: Cite at least one example from the literature for each to help your reader out or prune this sentence if you will discuss more fully in the section.
